homeassistant/components/sia/alarm_control_panel.py

```diff
@@ -0,0 +1,287 @@
+    EVENT_TIMESTAMP,
+    EVENT_ZONE,
+    PING_INTERVAL_MARGIN,
+    SIA_EVENT,
+)
+from .utils import get_attr_from_sia_event, get_entity_and_name, get_ping_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+DEVICE_CLASS_ALARM = "alarm"
+PREVIOUS_STATE = "previous_state"
+
+CODE_CONSEQUENCES = {
+    "PA": STATE_ALARM_TRIGGERED,
+    "JA": STATE_ALARM_TRIGGERED,
+    "TA": STATE_ALARM_TRIGGERED,
+    "BA": STATE_ALARM_TRIGGERED,
+    "CA": STATE_ALARM_ARMED_AWAY,
+    "CG": STATE_ALARM_ARMED_AWAY,
+    "CL": STATE_ALARM_ARMED_AWAY,
+    "CP": STATE_ALARM_ARMED_AWAY,
+    "CQ": STATE_ALARM_ARMED_AWAY,
+    "CS": STATE_ALARM_ARMED_AWAY,
+    "CF": STATE_ALARM_ARMED_CUSTOM_BYPASS,
+    "OA": STATE_ALARM_DISARMED,
+    "OG": STATE_ALARM_DISARMED,
+    "OP": STATE_ALARM_DISARMED,
+    "OQ": STATE_ALARM_DISARMED,
+    "OR": STATE_ALARM_DISARMED,
+    "OS": STATE_ALARM_DISARMED,
+    "NC": STATE_ALARM_ARMED_NIGHT,
+    "NL": STATE_ALARM_ARMED_NIGHT,
+    "BR": PREVIOUS_STATE,
+    "NP": PREVIOUS_STATE,
+    "NO": PREVIOUS_STATE,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: Callable[[], None]
+) -> bool:
+    """Set up SIA alarm_control_panel(s) from a config entry."""
+    async_add_entities(
+        [
+            SIAAlarmControlPanel(
+                *get_entity_and_name(
+                    entry.data[CONF_PORT], acc[CONF_ACCOUNT], zone, DEVICE_CLASS_ALARM
+                ),
+                entry.data[CONF_PORT],
+                acc[CONF_ACCOUNT],
+                zone,
+                acc[CONF_PING_INTERVAL],
+            )
+            for acc in entry.data[CONF_ACCOUNTS]
+            for zone in range(1, acc[CONF_ZONES] + 1)
+        ]
+    )
+    return True
+
+
+class SIAAlarmControlPanel(AlarmControlPanelEntity, RestoreEntity):
+    """Class for SIA Alarm Control Panels."""
+
+    def __init__(
+        self,
+        entity_id: str,
+        name: str,
+        port: int,
+        account: str,
+        zone: int,
+        ping_interval: int,
+    ):
+        """Create SIAAlarmControlPanel object."""
+        self.entity_id = ALARM_FORMAT.format(entity_id)
+        self._unique_id = entity_id
+        self._name = name
+        self._port = port
+        self._account = account
+        self._zone = zone
+        self._ping_interval_int = ping_interval
+        self._ping_interval = None
+        self._event_listener_str = f"{SIA_EVENT}_{port}_{account}"
+        self._unsub = None
+
+        self._is_available = True
+        self._remove_unavailability_tracker = None
+        self._state = None
+        self._old_state = None
+        self._attr = {
+            CONF_ACCOUNT: self._account,
+            CONF_PING_INTERVAL: self.ping_interval,
+            CONF_ZONE: self._zone,
+            EVENT_ACCOUNT: None,
+            EVENT_CODE: None,
+            EVENT_ID: None,
+            EVENT_ZONE: None,
+            EVENT_MESSAGE: None,
+            EVENT_TIMESTAMP: None,
```

Comment:
Our standard is to not include attributes in `extra_state_attributes` if they are `None`